discovery-provider/src/tasks/index.py

```diff
@@ -607,7 +612,16 @@ def index_blocks(self, db, blocks_list):
                 )
 
                 track_lexeme_state_changed = user_state_changed or track_state_changed
-                session.commit()
+                try:
+                    session.commit()
```

Comment:
confirming my recollection here - in the scenario where no consensus is reached with other discovery nodes failing to commit (i.e. a genuine local DB failure), this will be retried until either all others have encountered the same failure or the db commit succeeds. is that right?

Reply:
that's correct